agent-control/src/sub_agent/on_host/command/command_os.rs

```diff
@@ -122,58 +137,56 @@ impl CommandOSStarted {
 
         Ok(self)
     }
-}
-
-#[cfg(target_family = "unix")]
-mod unix {
-    use tracing::warn;
-
-    use crate::sub_agent::on_host::command::{command_os::CommandOSStarted, error::CommandError};
-
-    use std::time::Duration;
-    const POLL_INTERVAL: Duration = Duration::from_millis(100);
-
-    impl CommandOSStarted {
-        pub fn shutdown(&mut self) -> Result<(), CommandError> {
-            let pid = self.get_pid() as i32;
 
-            use nix::{sys::signal, unistd::Pid};
-            let graceful_shutdown_result = signal::kill(Pid::from_raw(pid), signal::SIGTERM)
-                .inspect_err(|err| warn!(agent_id = %self.agent_id, "Failed to gracefully exit process {pid}: {err}. Attempting forceful shutdown"));
+    fn is_running_after_timeout(&mut self, timeout: Duration) -> bool {
+        let deadline = Instant::now() + timeout;
 
-            if graceful_shutdown_result.is_err()
-                || self.is_running_after_timeout(self.shutdown_timeout)
-            {
-                self.process.kill().map_err(CommandError::from)?;
+        while Instant::now() < deadline {
+            if self.is_running() {
+                std::thread::sleep(POLL_INTERVAL);
+            } else {
+                return false;
             }
```

Comment:
Nit:(although I believe this hasn't actually changed)


```suggestion
            if !self.is_running() {
                return false;
            }
            std::thread::sleep(POLL_INTERVAL);
}
```

I think it is a bit more readable, is it just me?

Reply:
https://github.com/newrelic/newrelic-agent-control/pull/1862/commits/d77c6bd6da4a27cac5bfe36c472929b071d5369a